extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -142,6 +143,49 @@ function () {
 		return $data;
 	}
 
+
+	/**
+	 * Get $limit chat moderators who have contributed most recently
+	 * filter out bots and admins
+	 *
+	 * @param $limit
+	 * @return array
+	 */
+	public function getChatModerators( $limit ) {
+		$data = WikiaDataAccess::cache(
+			self::getMemcKey( [ self::MODERATORS_MCACHE_KEY, $limit ] ),
+			WikiaResponse::CACHE_STANDARD,
+			function () use ($limit) {
+				$db = wfGetDB( DB_SLAVE );
+
+				$moderatorIds = $this->wikiService->getWikiChatmoderatorIds( 0, false, true, null );
+				$adminIds = $this->wikiService->getWikiAdminIds( 0, false, true, null, true );
+				$botIds = $this->getBotIds();
+				$dateTwoYearsAgo = date( 'Y-m-d', strtotime( '-2 years' ) );
+
+				$sqlData = ( new WikiaSQL() )
+					->SELECT( 'distinct rev_user' )
+					->FROM( 'revision' )
+					->WHERE( 'rev_user' )->NOT_EQUAL_TO( 0 )
+					->AND_( 'rev_user' )->NOT_IN( $moderatorIds )
+					->AND_( 'rev_user' )->NOT_IN( $adminIds )
+					->AND_( 'rev_user' )->NOT_IN( $botIds )
+					->AND_( 'rev_timestamp' )->GREATER_THAN( $dateTwoYearsAgo )
+					->ORDER_BY( 'rev_timestamp DESC' )
+					->LIMIT( $limit );
+
+				$result = $sqlData->runLoop( $db, function ( &$result, $row ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)